core/app/models/spree/shipping_method.rb

```diff
@@ -15,6 +15,9 @@ class ShippingMethod < Spree::Base
     has_many :shipping_method_zones
     has_many :zones, through: :shipping_method_zones
 
+    has_many :shipping_method_stock_locations, class_name: Spree::ShippingMethodStockLocation
```

Comment:
As Ben says, this should be `dependent: :destroy`. Also, `class_name` should always be a string.